netmiko/mikrotik/mikrotik_ssh.py

```diff
@@ -0,0 +1,115 @@
+from netmiko.cisco_base_connection import CiscoSSHConnection
+
+
+class MikrotikBase(CiscoSSHConnection):
+    """Common Methods for Mikrotik RouterOS and SwitchOS"""
+
+    def __init__(self, **kwargs):
+        if kwargs.get("default_enter") is None:
+            kwargs["default_enter"] = "\r\n"
+
+        self.in_config_mode = True
+
+        return super(MikrotikBase, self).__init__(**kwargs)
+
+    def session_preparation(self, *args, **kwargs):
+        """Prepare the session after the connection has been established."""
+        self.ansi_escape_codes = True
+        # Clear the read buffer
+        self.write_channel(self.RETURN)
+        self.set_base_prompt()
+
+    def _modify_connection_params(self):
+        """Append login options to username
+        c: disable console colors
+        e: enable dumb terminal mode
+        t: auto detect terminal capabilities
+        w511: set term width
+        h4098: set term height
+        """
+        self.username += "+cetw511h4098"
+
+    def _enter_shell(self):
+        """Already in shell."""
+        return ""
```

Comment:
Shouldn't be needed--eliminate.

Reply:
Agreed, and in testing works as expected